File: ToolDemo/app/src/main/java/com/lilanz/tooldemo/daos/BeanDao.java
package com.lilanz.tooldemo.daos;

import android.annotation.SuppressLint;
import android.content.Context;
import android.util.ArrayMap;

import com.j256.ormlite.dao.Dao;

import java.sql.SQLException;
import java.util.List;

public class BeanDao<T> {

    private Dao<T, Integer> dao;

    @SuppressLint("NewApi")
    private static ArrayMap<Class, BeanDao> daoOperateMap = new ArrayMap<>();

    public synchronized static BeanDao getDaoOperate(Context context, Class clazz){
        if (!daoOperateMap.containsKey(clazz)){
            BeanDao beanDao = new BeanDao(context, clazz);
            daoOperateMap.put(clazz, beanDao);
        }
        return daoOperateMap.get(clazz);
    }

    private BeanDao(Context context, Class<T> clazz){
        dao = OrmLiteHelter.getInstance(context).getDao(clazz);
    }

    // 查询所有数据
    public List<T> queryAll(){
        List<T> beanList = null;
        try {
            synchronized (OrmLiteHelter.DATA_LOCK){
                beanList = dao.queryForAll();
            }
        }catch (SQLException ex){
            ex.printStackTrace();
        }
        return beanList;
    }

    // 插入一个数据对象
    public void insert(T t){
        try {
            synchronized (OrmLiteHelter.DATA_LOCK) {
                dao.createIfNotExists(t);
            }
        }catch (SQLException ex){
            ex.printStackTrace();
        }
    }

    // 删除数据对象
    public void delete(T t){
        try {
            synchronized (OrmLiteHelter.DATA_LOCK) {
                dao.delete(t);
            }
        }catch (SQLException ex){
            ex.printStackTrace();
        }
    }

    public void deleteById(int id){
        try {
            synchronized (OrmLiteHelter.DATA_LOCK) {
                dao.deleteById(id);
            }
        }catch (SQLException ex){
            ex.printStackTrace();
        }
    }

    // 插入数据集合
    public void insertList(List<T> list){
        try {
            synchronized (OrmLiteHelter.DATA_LOCK) {
                dao.create(list);
            }
        }catch (SQLException ex){
            ex.printStackTrace();
        }
    }

    // 更新数据
    public void update(T t){
        try {
            synchronized (OrmLiteHelter.DATA_LOCK) {
                dao.createOrUpdate(t);
            }
        } catch (SQLException ex) {
            ex.printStackTrace();
        }
    }

    // 删除数据集合
    public void deleteList(List<T> list){
        try {
            synchronized (OrmLiteHelter.DATA_LOCK) {
                dao.delete(list);
            }
        }catch (SQLException ex){
            ex.printStackTrace();
        }
    }
}

File: KotlinTool/app/src/main/java/com/lilanz/kotlintool/daos/BeanDao.kt
package com.lilanz.kotlintool.daos

import android.annotation.SuppressLint
import android.content.Context
import android.util.ArrayMap
import com.j256.ormlite.dao.Dao
import java.sql.SQLException
import java.util.*
import kotlin.collections.HashMap

class BeanDao<T> {
    companion object {
        @SuppressLint("NewApi")
        private val daoOperateMap: ArrayMap<Class<*>, BeanDao<*>?> = ArrayMap()

        @Synchronized
        fun getDaoOperate(context: Context, clazz: Class<*>): BeanDao<*>? {
            if (!daoOperateMap.containsKey(clazz)) {
                val beanDao = BeanDao(context, clazz)
                daoOperateMap[clazz] = beanDao
            }
            return daoOperateMap[clazz]
        }
    }


    private var dao: Dao<T, Int>? = null

    constructor(context: Context, clazz: Class<T>) {
        dao = OrmLiteHelter.getInstance(context)?.getDao(clazz)
    }

    // 查询所有数据
    fun queryAll(): List<T>? {
        var beanList: List<T>? = null
        try {
            synchronized(OrmLiteHelter.DATA_LOCK) { beanList = dao!!.queryForAll() }
        } catch (ex: SQLException) {
            ex.printStackTrace()
        }
        return if (beanList == null) ArrayList() else beanList
    }

    // 插入一个数据对象
    fun insert(t: T) {
        try {
            synchronized(OrmLiteHelter.DATA_LOCK) { dao!!.createIfNotExists(t) }
        } catch (ex: SQLException) {
            ex.printStackTrace()
        }
    }

    // 删除数据对象
    fun delete(t: T) {
        try {
            synchronized(OrmLiteHelter.DATA_LOCK) { dao!!.delete(t) }
        } catch (ex: SQLException) {
            ex.printStackTrace()
        }
    }

    fun deleteById(id: Int) {
        try {
            synchronized(OrmLiteHelter.DATA_LOCK) { dao!!.deleteById(id) }
        } catch (ex: SQLException) {
            ex.printStackTrace()
        }
    }

    // 插入数据集合
    fun insertList(list: List<T>?) {
        try {
            synchronized(OrmLiteHelter.DATA_LOCK) { dao!!.create(list) }
        } catch (ex: SQLException) {
            ex.printStackTrace()
        }
    }

    // 更新数据
    fun update(t: T) {
        try {
            synchronized(OrmLiteHelter.DATA_LOCK) { dao!!.createOrUpdate(t) }
        } catch (ex: SQLException) {
            ex.printStackTrace()
        }
    }

    // 删除数据集合
    fun deleteList(list: List<T>?) {
        try {
            synchronized(OrmLiteHelter.DATA_LOCK) { dao!!.delete(list) }
        } catch (ex: SQLException) {
            ex.printStackTrace()
        }
    }

    // 按条件查询第一条数据
    fun queryWhereForOne(map: HashMap<String, Any>): T? {
        if (map == null || map.size == 0) {
            return null
        }
        var t: T? = null
        try {
            synchronized(OrmLiteHelter.DATA_LOCK) {
                var where = dao!!.queryBuilder().where()
                var index = 0
                for ((key, value) in map) {
                    where = if (index == 0) {
                        where.eq(key, value)
                    } else {
                        where.and().eq(key, value)
                    }
                    index++
                }
                t = where.queryForFirst()
            }
        } catch (ex: SQLException) {
            ex.printStackTrace()
        }
        return t
    }

    // 按条件查询
    fun queryWhereForList(map: Map<String?, Any?>): List<T>? {
        var list: List<T>? = null
        try {
            synchronized(OrmLiteHelter.DATA_LOCK) {
                var where = dao!!.queryBuilder().where()
                var index = 0
                for ((key, value) in map) {
                    where = if (index == 0) {
                        where.eq(key, value)
                    } else {
                        where.and().eq(key, value)
                    }
                    index++
                }
                list = where.query()
            }
        } catch (ex: SQLException) {
            ex.printStackTrace()
        }
        return if (list == null) ArrayList() else list
    }
}